### PR TITLE
Mesh joining for unstable Bullet collision compounds.

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

Related to Issue #301 

Blocked by: PRs #309, #310. (diff includes these)

This PR addresses remaining physics instabilities when loading hierarchal meshes as Bullet rigid object by introducing optional but default mesh joining for more stable simulation of compound objects.

### Mesh joining:
Creating compound shapes as combinations of convex components can be an effective way of introducing macro concavities efficiently into simulation. PR #310 added this functionality by computing convex hulls for each mesh in a loaded collision mesh hierarchy and combining them into a single `btCompoundShape` object. However, when individual convex hulls overlap significantly, this treatment can result in unstable simulation artifacts. 
**This is demonstrated by simulation of a compound box shape with 2 convex sub-boxes only slightly offset from one another:** 
cube_compound_jitter_example
To remedy this, meshes can be joined into one convex hull, losing any concavity for improved stability and efficiency:
joined_cubecube

The mesh join option is added to the physics object config with default true for stability.

## How Has This Been Tested

Local testing.

Simulating random hierarchal objects after mesh joining and COM shift:
joined_convex_test_short

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/assets/MeshMetaData.h [hunk 1]
```diff
@@ -5,10 +5,46 @@
...
+
+  //! Object ID in the original file
+  int componentID;
+
+  std::vector<MeshTransformNode> children;
+
+  //! Node local transform
+  Magnum::Matrix4 T;
+
+  MeshTransformNode() {
```
Comment (msbaines) [inline src/esp/assets/MeshMetaData.h]: Why is this constructor needed? If you omit it, you can't create an undefined object and trying to create such an object will error at compile time.

File: src/esp/assets/MeshMetaData.h [hunk 2]
```diff
@@ -5,10 +5,46 @@
...
+  //! material ID within @ref MeshMetaData::materialIndex
+  int materialIDLocal;
+
+  //! Object ID in the original file
+  int componentID;
+
+  std::vector<MeshTransformNode> children;
+
+  //! Node local transform
+  Magnum::Matrix4 T;
```
Comment (mosra) [inline src/esp/assets/MeshMetaData.h]: Can we please *not* make variables with an uppercase first letter? :pray: It's making the code hard to read, especially with `T`, which is often a template parameter indicating a *type*.

File: src/esp/assets/MeshMetaData.h [hunk 3]
```diff
@@ -5,10 +5,46 @@
...
+  MeshTransformNode(const MeshTransformNode& val) {
+    componentID = val.componentID;
+    meshIDLocal = val.meshIDLocal;
+    materialIDLocal = val.materialIDLocal;
+    T = Magnum::Matrix4(val.T);
+    for (auto& child : val.children) {
+      children.push_back(MeshTransformNode(child));
+    }
+  }
+};
```
Comment (mosra) [inline src/esp/assets/MeshMetaData.h]: This is just a copy constructor, right? Why not let the compiler generate it automatically?

File: src/esp/assets/ResourceManager.cpp [hunk 4]
```diff
@@ -1034,43 +1013,31 @@ void ResourceManager::loadTextures(Importer& importer, MeshMetaData* metaData) {
...
-                                   int componentID) {
-  std::unique_ptr<Magnum::Trade::ObjectData3D> objectData =
-      importer.object3D(componentID);
-  if (!objectData) {
-    LOG(ERROR) << "Cannot import object " << importer.object3DName(componentID)
-               << ", skipping";
-    return;
-  }
-
+                                   MeshTransformNode& meshTransformNode) {
```
Comment (mosra) [inline src/esp/assets/ResourceManager.cpp]:
```suggestion
                                   const MeshTransformNode& meshTransformNode) {
```

I don't see why this should be a mutable reference.

**Comment by aclegg3** [comment]
> @aclegg3 what's the status of this PR, is it good to land?

Waiting on C++ test coverage for new functions. Functionality is fine. I'll look into this soon.

**Pull request merged**